### DISCUSSION TITLE
Some confused about speech to text

### DISCUSSION THREAD

**Discussion by niltor** [CommunityToolkit/Maui#1210] [Q&A] [answered]
This Interface just have one method `ListenAsync`, I had tried use it and I have some questions.
## Sample codes 
```csharp
var recognitionResult = await speechToText.ListenAsync(
                                            CultureInfo.GetCultureInfo(locale?. Language ?? defaultLanguage),
                                            new Progress<string>(partialText =>
                                            {

                                            }), cancellationToken);
if (recognitionResult.IsSuccessful)
{

}
```

## Questions
1. If I keep talking, I will always wait for the ListenAsync method to complete and never get a Result？
2. If I give a cancellationToken, so no more recognitionResult?
3. When should I process the resulting text? After ListenAsync task done, or in the IProcess stage?
4. Is it possible to configure the silence timeout (to know conversation is over)
5. Does the converted text contain punctuation, or is it configurable?

## My Scenes
When we have `speech to text` and `text to speech` ability on device, and at the same time we have ChatGPT,
I just need to use OpenAI chat API to get the response, so we can use voice chat instead of text chat.

So, I need to get the converted text (recognizable into a sentence) and use the OpenAI API to generate a dialogue in near real time, and then let the device read it out.

Anyone have any suggestion?

**Comment by VladislavAntonyuk** [comment, marked as answer]
Example with chat gpt. https://github.com/VladislavAntonyuk/MauiSamples/tree/main/MauiSpeech

1.  Yes.
2. if operation cancels you only have result from IProgress
3. you can put code in IProgress action and get intermediate results.
4. Currently no. But it’s a good idea.
5. Yes. Comma is converted to ,

**Comment by CliffAgius** [comment]
Closed as answered
